fabric-message-api-v1/src/main/java/net/fabricmc/fabric/api/message/v1/ServerChatEvents.java

```diff
@@ -0,0 +1,174 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.message.v1;
+
+import net.minecraft.network.MessageType;
+import net.minecraft.network.encryption.SignedChatMessage;
+import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.server.filter.Message;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.Text;
+import net.minecraft.util.registry.RegistryKey;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+public final class ServerChatEvents {
+	private ServerChatEvents() {
+	}
+
+	/**
+	 * An event that is called when a player sends a chat message to all players, either via the
+	 * chat hud or the {@code /say} command. Mods can use this event to block the message.
+	 */
+	public static final Event<AllowChatMessage> ALLOW_CHAT_MESSAGE = EventFactory.createArrayBacked(AllowChatMessage.class, handlers -> (message, sender, typeKey) -> {
+		for (AllowChatMessage handler : handlers) {
+			if (!handler.allowChatMessage(message, sender, typeKey)) return false;
+		}
+
+		return true;
+	});
+
+	/**
+	 * An event that is called when the server sends a game message to all players. Game messages
+	 * include death messages, join/leave messages, and advancement messages. Mods can use this
+	 * event to block the message.
+	 */
+	public static final Event<AllowGameMessage> ALLOW_GAME_MESSAGE = EventFactory.createArrayBacked(AllowGameMessage.class, handlers -> (message, typeKey) -> {
+		for (AllowGameMessage handler : handlers) {
+			if (!handler.allowGameMessage(message, typeKey)) return false;
+		}
+
+		return true;
+	});
+
+	/**
+	 * An event that is called when a command sends a message to all players. Note that if a player
+	 * executed the command, {@link #ALLOW_CHAT_MESSAGE} will also be triggered. To check whether the
+	 * player executed the command, use {@code source.getPlayer() != null}. Mods can use this event
+	 * to block the message.
+	 */
+	public static final Event<AllowCommandMessage> ALLOW_COMMAND_MESSAGE = EventFactory.createArrayBacked(AllowCommandMessage.class, handlers -> (message, source, typeKey) -> {
+		for (AllowCommandMessage handler : handlers) {
+			if (!handler.allowCommandMessage(message, source, typeKey)) return false;
+		}
+
+		return true;
+	});
+
+	/**
+	 * An event that is called when a player sends a chat message to all players, either via the
+	 * chat hud or the {@code /say} command.
```

Comment:
chat-hut is unclear, makes it sound like the server gui has something, while it probably refers to chat messages received from the client as typically entered through the chat hud/gui.

send<->allow relation should be documented on both.

not including the `SEND_` prefix in these names could be better, the current ones with the verb sound a lot like something wsill be actively triggered vs getting a passive notification while not adding much.